compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -474,7 +474,9 @@ fn construct_fn<'tcx>(
     };
 
     let mut abi = fn_sig.abi;
-    if let DefKind::Closure = tcx.def_kind(fn_def) {
+    if let DefKind::Closure = tcx.def_kind(fn_def)
+        && !tcx.is_coroutine(fn_def)
```

Comment:
This concerns me, since I hope that coroutines don't have `abi::RustCall`.

Reply:
Previously, `﻿DefKind` included both ﻿`DefKind::Closure` and `﻿DefKind::Coroutine`. This change is simply to ensure that they are treated the same.